### PR TITLE
string class #955

### PR DESCRIPTION
**Pull request by alfabeat**
## Pull Request Type

What kind of change does this PR introduce?

[ ] Solved an issue/task

## What is the current behavior?
string class

Issue Number: [#955 ]

## What is the new behavior?

added string class

### CODE REVIEW

File: tasks/c++/hard/mystring.cpp [hunk 1]
```diff
@@ -24,32 +46,78 @@ MyString::~MyString() {
...
     // TODO: Implement this
+    MyString temp;
+    temp.len = (len + other.len)+1;
+    temp.str = new char[temp.len];
+    strcpy(temp.str, str);
+    strcat(temp.str, other.str);
+    temp.str[len] = '\0';
+    return temp;
+
 }
```
Comment (nikohoffren) [inline tasks/c++/hard/mystring.cpp:102]: The class looks mostly good but it does not quite work. It prints only `Hello` when executing, and i think the problem might be not using proper memory allocation for temp.str in the operator+ method, which will cause it to crash. If you can fix that, i can merge this PR.

**Review by nikohoffren — CHANGES_REQUESTED** [review]

**Comment by nikohoffren** [comment]
Just as a reminder, bot will close this PR if yoi don't respond to the review in two days from now.

**Comment by alfabeat** [comment]
thank you for the reminder

**Review by nikohoffren — APPROVED** [review]
Thanks! The program seems to be working as expected now.

**Comment by nikohoffren** [comment]
## Merged

> This is an automated message from Fork, Commit, Merge [BOT].

Thank you for your contribution! Your pull request has been merged. The files have been reset for the next contributor.

### What's next?

If you're looking for more ways to contribute, I invite you to check out my other projects. Just click [here](https://github.com/nikohoffren?tab=repositories) to find more. These projects contain real issues that you can help resolve. You can also check out the [Influences](https://github.com/nikohoffren/fork-commit-merge#influences) section in the README to find more projects similar to this one.

Also please leave a star to this project if you feel it helped you, i would really appreciate it.

I look forward to seeing your contributions!

**Pull request merged**